### PR TITLE
Document the export user data command.

### PR DESCRIPTION
**Pull request by thezaidbintariq**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by thezaidbintariq** [comment]
Sign-off-by: Zaid Bin Tariq ztariq.bscs19seecs@seecs.edu.pk

**Comment by clokep** [comment]
Could you please add a change log file? It can be created as `changelog.d/14883.doc`. Thanks!

**Comment by thezaidbintariq** [comment]
> Could you please add a change log file? It can be created as `changelog.d/14883.doc`. Thanks!

#14888 everything good now?

**Comment by dklimpel** [comment]
You have to add the newsfile to this PR and not a separate PR.

**Comment by thezaidbintariq** [comment]
> You have to add the newsfile to this PR and not a separate PR.
is it good now?

**Comment by dklimpel** [comment]
> is it good now?

Yes.

File: docs/usage/administration/admin_faq.md [hunk 1]
```diff
@@ -32,6 +32,12 @@ What users are registered on my server?
 SELECT NAME from users;
 ```
 
+How can I export user data?
+---
+Export data of specific user by running a python script that exports data and takes in user_id and config file as arguments.
+```python -m synapse.app.admin_cmd -c <config_file> export-data <user_id>
```
Comment (dklimpel) [inline docs/usage/administration/admin_faq.md:38]:
```suggestion
```console
python -m synapse.app.admin_cmd -c <config_file> export-data <user_id>
```

File: changelog.d/14883.doc [hunk 2]
```diff
@@ -0,0 +1 @@
+Added command to export user data. Contributed by @thezaidbintariq
```
Comment (dklimpel) [inline changelog.d/14883.doc:1]:
```suggestion
Added command to export user data. Contributed by @thezaidbintariq.
```

**Review by dklimpel — CHANGES_REQUESTED** [review]

**Comment by dklimpel** [comment]
Closes
- #14530

**Comment by dklimpel** [comment]
IMHO there is also an optional parameter `--output-directory` for the directory.

https://github.com/matrix-org/synapse/blob/22cc93afe38d34c859d8863a99996e7e72ca1733/synapse/app/admin_cmd.py#L214-L217

**Comment by dklimpel** [comment]
https://github.com/matrix-org/synapse/blob/22cc93afe38d34c859d8863a99996e7e72ca1733/.ci/scripts/test_export_data_command.sh#L21-L23

This is an example how to run with `poetry`.

File: docs/usage/administration/admin_faq.md [hunk 3]
```diff
@@ -32,6 +32,13 @@ What users are registered on my server?
 SELECT NAME from users;
 ```
 
+How can I export user data?
+---
+Export data of specific user by running a python script that exports data and takes in user_id and config file as arguments.
```
Comment (clokep) [inline docs/usage/administration/admin_faq.md:37]:
```suggestion
Synapse includes a Python command to export data for a specific user. It takes the homeserver
configuration file and the full Matrix ID of the user to export:
```

File: changelog.d/14883.doc [hunk 4]
```diff
@@ -0,0 +1 @@
+Added command to export user data. Contributed by @thezaidbintariq.
```
Comment (clokep) [inline changelog.d/14883.doc:1]:
```suggestion
Document the export user data command. Contributed by @thezaidbintariq.
```

**Pull request merged**